pennylane/decomposition/utils.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+This module implements utility functions for the decomposition module.
+
+"""
+
+
+class DecompositionError(Exception):
+    """Base class for decomposition errors."""
+
+
+def toggle_graph_decomposition():
+    """A closure that toggles the experimental graph-based decomposition on and off."""
+
+    _GRAPH_DECOMPOSITION = False
+
+    def enable():
+        """
+        A global toggle for enabling the experimental graph-based decomposition system
+        in PennyLane (introduced in v0.41). This new way of doing decompositions is
+        generally more performant and allows for specifying custom decompositions.
+
+        When this is enabled, `qml.transforms.decompose` will use the new decompositions system.
+        """
+
+        nonlocal _GRAPH_DECOMPOSITION
+        _GRAPH_DECOMPOSITION = True
+
+    def disable() -> None:
+        """
+        A global toggle for disabling the experimental graph-based decomposition
+        system in PennyLane (introduced in v0.41). The experimental graph-based
+        decomposition system is disabled by default in PennyLane.
+
+        See also: :func:`decomposition.enable_graph`
+        """
+
+        nonlocal _GRAPH_DECOMPOSITION
+        _GRAPH_DECOMPOSITION = False
+
+    def status() -> bool:
+        """
+        A global toggle for checking the status of the experimental graph-based
+        decomposition system in PennyLane (introduced in v0.41). The experimental
+        graph-based decomposition system is disabled by default in PennyLane.
+
+        See also: :func:`decomposition.enable_graph`
+        """
+
+        nonlocal _GRAPH_DECOMPOSITION
+        return _GRAPH_DECOMPOSITION
+
+    return enable, disable, status
+
+
+enable_graph, disable_graph, enabled_graph = toggle_graph_decomposition()
+"""
+The following functions are used to enable and disable the graph-based decomposition algorithm:
+- enable_graph: Enables the graph-based decomposition algorithm.
+- disable_graph: Disables the graph-based decomposition algorithm.
+- enabled_graph: Returns the current status of the graph-based decomposition algorithm.
+"""
```

Comment:
This shows up weirdly in the docs 🤔 

![Screenshot 2025-03-20 at 5 28 28 PM](https://github.com/user-attachments/assets/c8562fcf-2bf9-4f03-b2c2-27b29166acfa)

Reply:
I think it's fixed now [e7b3a4d](https://github.com/PennyLaneAI/pennylane/pull/6966/commits/e7b3a4d3a1469453fbe7623c58bf7f3120258ffd) :crossed_fingers: